### PR TITLE
Non-curated warning

### LINKED ISSUE #935: Warn on installing non-curated apps
Similar to #808, we should warn users that non-curated apps are not reviewed or approved by elementary, so any guarantees we make (i.e. in the Privacy blog post) do not apply. 

This was pointed out by someone on Reddit in response to the blog post.

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/68700667-warn-on-installing-non-curated-apps?utm_campaign=plugin&utm_content=tracker%2F57667267&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F57667267&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by cassidyjames**
Fixes #935 

- [x] Consolidate the two Content Warning checks to one method
- [x] Add new Non-Curated Warning dialog
- [x] Add non-curated warning GSetting
- [x] Clean up Content Warning dialog for style and to be easier to work with
- [x] Decide what to focus by default
- [ ] Suggested or destructive action?

Screenshot from 2019-10-24 20 24 28@2x

### CODE REVIEW

**Comment by cassidyjames** [comment]
Current status:

image

File: src/Dialogs/ContentWarningDialog.vala [hunk 1]
```diff
@@ -29,8 +27,8 @@ public class AppCenter.Widgets.ContentWarningDialog : Granite.MessageDialog {
     }
 
     construct {
-        primary_text = _("%s Contains Explicit Content").printf (app_name);
-        secondary_text = _("%s may contain content only suitable for adults or that may be sensitive or disturbing. In general, elementary does not review or endorse the contents of this app.").printf (app_name);
+        primary_text = _("“%s” contains explicit content").printf (app_name);
+        secondary_text = _("“%s” may contain content only suitable for adults or that may be sensitive or disturbing. In general, elementary does not review or endorse the contents of this app.").printf (app_name);
```
Comment (lainsce) [inline src/Dialogs/ContentWarningDialog.vala]: Don't forget to remove the printf that does nothing now :P

File: src/Dialogs/NonCuratedWarningDialog.vala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        Object (
+            app_name: _app_name,
+            image_icon: new ThemedIcon ("dialog-warning"),
+            title: _("Non-Curated Warning")
+        );
+    }
+
+    construct {
+        primary_text = _("“%s” is not curated").printf (app_name);
+        secondary_text = _("This app is not distributed or updated by elementary. Its listing, contents, and functionality have not been reviewed or approved. Install at your own risk.");
```
Comment (lainsce) [inline src/Dialogs/NonCuratedWarningDialog.vala]: Don't forget to remove the `printf()` that does nothing :P

**Comment by cassidyjames** [comment]
Updated design and copy inspired by [Sideload](https://github.com/elementary/sideload) after talking with @danrabbit 


image (1)

**Comment by cassidyjames** [comment]
Using `security-low` icon

Screenshot from 2019-10-25 12 44 26@2x

**Pull request merged**